pkg/kubectl/cmd/clusterinfo.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+
+	"github.com/spf13/cobra"
+)
+
+func (f *Factory) NewCmdClusterInfo(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "clusterinfo",
+		Short: "Display cluster info",
+		Long:  "Display IPs of the master and services with label kubernetes.io/cluster-service=true",
+		Run: func(cmd *cobra.Command, args []string) {
+			RunClusterInfo(f, out, cmd)
+		},
+	}
+	return cmd
+}
+
+func RunClusterInfo(f *Factory, out io.Writer, cmd *cobra.Command) {
+	client, err := f.ClientConfig(cmd)
+	checkErr(err)
+	fmt.Fprintf(out, "Kubernetes master is running at %v\n", client.Host)
+
+	mapper, typer := f.Object(cmd)
+	cmdNamespace, err := f.DefaultNamespace(cmd)
+	checkErr(err)
+	args := []string{"pods"}
+	// TODO: use generalized labels once they are implemented (#341)
+	b := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand(cmd)).
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		SelectorParam("kubernetes.io/cluster-service=true").
+		ResourceTypeOrNameArgs(false, args...).
+		Latest()
+
+	b.Do().Visit(func(r *resource.Info) error {
+		pods := r.Object.(*api.PodList).Items
+		for _, pod := range pods {
+			fmt.Fprintf(out, "%v is running at http://%v\n", pod.Labels["name"], pod.Status.HostIP)
```

Comment:
Ah, do you mean an answer to the question "What about ports? For example is
there any option to get a port of Kibana?"

Satnam

On Tue, Feb 24, 2015 at 5:58 AM, Piotr Szczesniak <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/clusterinfo.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4657#discussion_r25252969>
> :
>
> > +
> > +	mapper, typer := f.Object(cmd)
> > +	cmdNamespace, err := f.DefaultNamespace(cmd)
> > +	checkErr(err)
> > +	args := []string{"pods"}
> > +	// TODO: use generalized labels once they are implemented (#341)
> > +	b := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand(cmd)).
> > +		NamespaceParam(cmdNamespace).DefaultNamespace().
> > +		SelectorParam("kubernetes.io/cluster-service=true").
> > +		ResourceTypeOrNameArgs(false, args...).
> > +		Latest()
> > +
> > +	b.Do().Visit(func(r *resource.Info) error {
> > +		pods := r.Object.(*api.PodList).Items
> > +		for _, pod := range pods {
> > +			fmt.Fprintf(out, "%v is running at http://%v\n", pod.Labels["name"], pod.Status.HostIP)
>
> There is no hostPort for pods. I can hardcode them but it's not a good
> pattern. Using services also is not ideal solution (as discussed in #3290
> <https://github.com/GoogleCloudPlatform/kubernetes/issues/3290>) because
> not every cluster service is exported publicly.
>
> I asked about ports in this comment
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4417#discussion_r24994580>
> but haven't received any response.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4657/files#r25252969>
> .
>